### PR TITLE
Cleanup ranges test machinery

### PR DESCRIPTION
**Pull request by CaseyCarter**
Use helper function `to_bool` to replace function-style casts to `bool`. It's cleaner while still being more concise than a full `static_cast`.

### CODE REVIEW

File: tests/std/tests/P0896R4_ranges_subrange/test.cpp [hunk 1]
```diff
@@ -98,13 +98,13 @@ namespace test_view_interface {
...
-        using S = std::conditional_t<bool(IsCommon), I, test::sentinel<int>>;
+        using S = std::conditional_t<static_cast<bool>(IsCommon), I, test::sentinel<int>>;
 
         I begin();
-        I begin() const requires (bool(HasConstRange));
+        I begin() const requires (static_cast<bool>(HasConstRange));
 
         S end();
-        S end() const requires (bool(HasConstRange));
+        S end() const requires (static_cast<bool>(HasConstRange));
```
Comment (StephanTLavavej) [inline tests/std/tests/P0896R4_ranges_subrange/test.cpp:107]: Would it be possible to add `test::to_bool` to the using-declarations on line 94 and then call it here? Or is that stylistically undesirable, or affected by DevCom-1034539?

Comment (CaseyCarter) [reply tests/std/tests/P0896R4_ranges_subrange/test.cpp:107]: No, that should work fine.

Comment (CaseyCarter) [reply tests/std/tests/P0896R4_ranges_subrange/test.cpp:107]: Err, "Yes, that should work fine; no, the bug doesn't affect unqualified lookup, only argument-dependent lookup."

**Comment by StephanTLavavej** [comment]
Thanks for cleaning up these functional-style casts! To celebrate, here are functional-style cats: :smile_cat:(:smiley_cat:)

**Comment by CaseyCarter** [comment]
> Thanks for cleaning up these functional-style casts! To celebrate, here are functional-style cats: 😸(😺)

I think you've rediscovered Schrödinger's lambda calculus.

**Pull request merged**